airbyte-integrations/connectors/source-postgres/src/main/java/io/airbyte/integrations/source/postgres/PostgresSource.java

```diff
@@ -455,17 +462,91 @@ public List<AutoCloseableIterator<AirbyteMessage>> getIncrementalIterators(final
+          if (streamState == null || streamDescriptor == null) {
+            return;
+          }
+          final AirbyteStateMessage clonedState = Jsons.clone(s);
+          if (streamState.has("type") && streamState.get("type").asText().equalsIgnoreCase("ctid")) {
+            statesFromCtidSync.add(clonedState);
+            streamsStillInCtidSync.add(new AirbyteStreamNameNamespacePair(streamDescriptor.getName(), streamDescriptor.getNamespace()));
+          } else {
+            statesFromXminSync.add(clonedState);
+          }
+          alreadySeenStreams.add(new AirbyteStreamNameNamespacePair(streamDescriptor.getName(), streamDescriptor.getNamespace()));
+        });
+      }
+
+      final List<ConfiguredAirbyteStream> newlyAddedStreams = identifyNewlyAddedStreams(catalog, alreadySeenStreams);
+      final List<ConfiguredAirbyteStream> streamsForCtidSync = new ArrayList<>();
+      catalog.getStreams().stream()
+          .filter(stream -> streamsStillInCtidSync.contains(AirbyteStreamNameNamespacePair.fromAirbyteStream(stream.getStream())))
+          .map(Jsons::clone)
+          .forEach(streamsForCtidSync::add);
+
+      streamsForCtidSync.addAll(newlyAddedStreams);
+
+      final List<ConfiguredAirbyteStream> streamsForXminSync = catalog.getStreams().stream()
+          .filter(stream -> !streamsForCtidSync.contains(stream))
+          .map(Jsons::clone)
+          .toList();
+
+      LOGGER.info("Streams to be synced via ctid : {}", streamsForCtidSync.size());
+      LOGGER.info("Streams to be synced via xmin : {}", streamsForXminSync.size());
+
+      final List<AutoCloseableIterator<AirbyteMessage>> ctidIterator = new ArrayList<>();
+      final List<AutoCloseableIterator<AirbyteMessage>> xminIterator = new ArrayList<>();
+
+      if (!streamsForCtidSync.isEmpty()) {
+        final CtidStateManager ctidStateManager = new CtidStateManager(statesFromCtidSync);
+        final PostgresCtidHandler ctidHandler = new PostgresCtidHandler(database, sourceOperations, getQuoteString(), ctidStateManager,
+            x -> new ObjectMapper().valueToTree(xminStatus),
+            (pair, jsonState) -> XminStateManager.getAirbyteStateMessage(pair, Jsons.object(jsonState, XminStatus.class)));
+        ctidIterator.addAll(ctidHandler.getIncrementalIterators(
+            new ConfiguredAirbyteCatalog().withStreams(streamsForCtidSync), tableNameToTable, emittedAt));
+      }
+
+      if (!streamsForXminSync.isEmpty()) {
+        final XminStateManager xminStateManager = new XminStateManager(statesFromXminSync);
+        final PostgresXminHandler xminHandler = new PostgresXminHandler(database, sourceOperations, getQuoteString(), xminStatus, xminStateManager);
+
+        xminIterator.addAll(xminHandler.getIncrementalIterators(
+            new ConfiguredAirbyteCatalog().withStreams(streamsForXminSync), tableNameToTable, emittedAt));
+      }
+
+      return Stream
```

Comment:
nit : More of an extension of the discussion we had in the AM

Maybe not for this PR, but this is the issue I was talking about : https://github.com/airbytehq/airbyte/issues/27115 where ideally we want to kick off an incremental xmin sync after the initial successful ctid sync. 

One way to do this is to slightly augment what you have here. Even for streams that belong to `streamsForCtidSync` we can build iterators for the xmin query and merge them so that the xmin iterator runs after the ctid iterator. 

When that happens, we'll have to make sure that the handlers can merge streams based on stream name